idaes/models/unit_models/mscontactor.py

```diff
@@ -558,15 +569,49 @@ def _build_heterogeneous_reaction_blocks(self):
                 "reactions (reaction_idx)."
             )
 
+    def _add_geometry(self, uom):
+        if self.config.has_holdup:
+            # Add volume for each element
+            # TODO: Assuming constant volume for now
+            self.volume = Var(
+                self.elements,
+                initialize=1,
+                units=uom.VOLUME,
+                doc="Volume of element",
+            )
```

Comment:
Consider if you want to index `self.volume` by time now if you have plans to make it time-varying eventually. Having an index that a variable is constant along is easier than trying to add the index after the fact.